dotnet/src/VectorDataIntegrationTests/VectorDataIntegrationTests/Support/TestStore.cs

```diff
@@ -26,7 +27,24 @@ public abstract class TestStore
     protected virtual Task StopAsync()
         => Task.CompletedTask;
 
-    public abstract VectorStore DefaultVectorStore { get; }
+    public VectorStore DefaultVectorStore
+    {
+        get => this._defaultVectorStore ?? throw new InvalidOperationException("Not initialized");
```

Comment:
This logic was repeated for all the derived types. I assume it was done to allow them to be able to use derived type rather than `VectorStore`. But it was not really used and I was getting a compiler warning about `IDisposable` field in every derived type, so I've just moved this logic (and the ownership) to the base class.